src/psyclone/psyGen.py

```diff
@@ -4705,6 +4968,300 @@ def _number_handler(self, node, parent):
         return Literal(node.items[0], parent=parent)
 
 
+class Symbol(object):
+    '''
+    Symbol item for the Symbol Table. It contains information about: the name,
+    the datatype, the shape (in row-major order), the scope and for
+    global-scoped symbols whether the data is already defined and/or survives
+    after the kernel.
+
+    :param str name: Name of the symbol.
+    :param str datatype: Data type of the symbol.
+    :param list shape: Shape of the symbol in row-major order (leftmost \
+                       index is contiguous in memory). Each entry represents \
+                       an array dimension. If it is 'None' the extent of that \
+                       dimension is unknown, otherwise it holds an integer \
+                       with the extent. If it is an empy list then the symbol \
+                       represents a scalar.
+    :param str scope: It is 'local' if the symbol just exists inside the \
+                      kernel scope or 'global_*' if the data survives outside \
+                      of the kernel scope. Note that global-scoped symbols \
+                      also have postfixed information about the sharing \
+                      mechanism, at the moment just 'global_argument' is \
+                      available for variables passed in/out of the kernel \
+                      by argument.
+    :param bool is_input: Whether it is an already existent data that is \
```

Comment:
"Whether the symbol represents data that exists before the kernel is entered and that is passed into the kernel." (and elsewhere please)

Reply:
Done